### DISCUSSION TITLE
Converting email to Task

### DISCUSSION THREAD

**Discussion by Papapooch** [stacks-task-manager/stacks2#45] [Ideas & Feature requests]
First, I am a happy Newbie, about to purchase based on what I already see and the consistent pace of fixes and improvements. But one of the few things I don't see yet (but may have missed) and is critical to me (and oft seen in other Task/Project solutions) is the ability to direct an email to  dedicated email address and have it slug in a new Task. Naturally, this requires some editing of the Task, but as I see it (after some 35 years into entrepreneurship in marketing technologies development). In the early days of Asana, they tried to convince Users that there would be a world where such ad hoc conversion of an email could be flipped into a Task, but they were wrong. Today, this functionality is in nearly every Task Manager, which I am sure is not news to you, Cris. I didn't see a reference to this functionality of the Roadmap. Is it a possibility soon? Between this function and some advancements in web content clipping for Notes and Tasks, you are very close to being at the top of a very tall "stack" of others - Stacks is clean, efficient, and privacy matters more and more every day. You are to be congratulated for your execution of the most elegant solution. I am looking forward to supporting it.

**Comment by synapse** [comment]
Hey @Papapooch, nope it isn't in the roadmap but I have it in my private to-do list. This feature was left there till the very end for good reason/s:
- the user will be required to enter a email's IMAP info, in order for the app to listen to incoming emails and convert them into task in case there's anything tagged in a specific way
- in order for this to work in realtime, that app should be running
- Stacks can work offline, so this might cause confusion if the task would not be created when the email is being received by the IMAP server

Overall this would be a great addition to the current features. Customers could easily benefit from this automation.

**Reply by Papapooch** [reply]
Got it. All true... both Stacks and email client would have to be running but is this one of those "if this then that" benefits? There is a valid argument for running offline and locally, on the other hand, this functionality is so very useful. As to ease of set-up which I know has to concern you, there is code available that makes setting up IMAP/SMTP nearly automatic. Critical, of course, because you sure don't want to be troubleshooting email server addresses, ports, and security protocols ;-) but it has gotten so much better lately. You can watch it happen beautifully over at BlueMail, Mailsring, Mailbird, and OnMail, for example. Again, I get it. I can't write a line of code even though I have spent millions with coders, was granted a Martech U.S. Patent, have launched a half-dozen solutions over 35 years, and have sent over one billion marketing/fundraising emails over 10 years but I know in my deepest core that some functional improvements are simpler than others but nothing about systems development, deployment, and support is ever "easy". It's all about ROI and the good discipline you need to succeed in systems development. That said, as a project/task management freak and multi-project developer, you show obvious chops. I look at Stacks and can't help but say that if you include this email-to-task conversion with the online proviso and you have met and even exceeded what has taken Todoist, TickTick, and dozens of others a decade plus to deliver. Nail down web content clipping that is well integrated with task management (how I found you) that gets rendered on the Kanban view and you have (for my money) overtaken Evernote and others who have spent years and (likely) millions to deliver tasks that intersect with content. Again, I get the local vs. hosted, privacy angle. The question is what that costs (the day-to-day User) and whether that eventually compromises your long-term value to the User. How many folks are really running exclusively locally? I know I can't because I move between a desktop and two notebooks. And a Mobile version eventually requires a web connection right? Now in my case, I intend to eventually park Stacks on a local, WIFI with a dedicated IP address NAS. Most would not be willing to go that route but methinks that should work. OK, enough of my ranting ;-) I want, above all else to encourage you. I think your work product is exceptional and it stands out in a very crowded field of contenders both dead and alive. Onward...

**Reply by synapse** [reply]
Overall this should not be a difficult task, but I'll have to do some investigation before adding this one. I also need to address a couple o functionality questions first, like:

- where should this be located: workspace or globally in the app
    - workspace wise: what happens when an email arrives and it's not the right workspace selected
    - app wise: where should the task be created? in which workspace, project, column
    - which of the above is better
    - should there be multiple IMAP accounts, one per workspace
- how much memory does the IMAP client consume while in listening more
